internal/configuration/schema/theme.go

```diff
@@ -0,0 +1,15 @@
+package schema
+
+// Theme represents the configuration related to Styling.
+type ThemeConfiguration struct {
+        Name                 string `mapstructure:"name"`
+        MainColor            string `mapstructure:"maincolor"`
+        SecondaryColor       string `mapstructure:"secondarycolor"`
+}
+
+// DefaultThemeConfiguration represents default configuration parameters for Theming/Styling.
+var DefaultThemeConfiguration = ThemeConfiguration{
+	Name: "light",
+	MainColor: "#1976d2", //Blue
```

Comment:
So right now anyone would be able to dynamically customize the primary and secondary colors and it's great but I'm wondering (open question) whether we'd really like to have every part of the theme dynamically customizable in the future (font? margins? paddings? borders? etc...). Where do we put the cursor?

I think stopping there is good as it will reduce the scope of what is dynamically customizable to the bare minimum to avoid introducing more code in the backend to support themes (KISS). This could obviously be challenged later but I'd rather keep the code minimal.

However, a user willing to introduce a customization to any part of it should be able do easily do it by creating a new theme easily and rebuild Authelia with it. It's a bit less user friendly but doable and it's also why the theme name brings value otherwise we could just skip the theme name and customize all variables assuming they would be dynamically customizable.

Let me know what you think.

Reply:
Also if the theme name and the colors are not mutually exclusive in the configuration, I'm not sure to understand the need for a `custom` theme. Can you clarify? I mean someone could take any theme as the base and customize the colors, what does `custom` brings here?
If `custom` is just another base theme, why not give it a proper self-explaining name like we did with `light` and `dark`.

Reply:
I think it is more maintainable to keep it simple. I would suggest to limit it to colors being fully customizable. As we could go further, but it seems to me that this, as you say, would only make the codebase more complicated than necessary But who knows, maybe there will be a need/envy to extend the theming option in the future with background svg's or else.

I liked the idea of having for any contributor, the ability to create a theme, do a pull request to implement his theme with a defined name, like "light" or "dark", to in the end have an easy way for new users, to change their layouts/colors with predefined/tested themes.

As you say, the other approach would be to have the default theme as of now, and just make both, primary and secondary color variables available for customisation. Without even providing any defined themes.

My idea with the custom theme was that users could try easily/quickly color combinations with just setting two variables in the conf, and that's it. It's really more oriented towards new users which would have not much idea on how to create a new theme, while at the same time having also pre-created themes at their disposal. (for inspiration for example).

I see in fact a change needed, in terms of, if custom is NOT selected then primary_color and secondary_color keys should not be allowed? As they would be ignored anyways.

Not sure you understand my logic with this. If something is still unclear, let me know and i'll try to be more specific.

Maybe you have some other approach/idea ?
Especially if we already consider how we would handle this once there is an admin/user space available for defining theming/unbanning etc.

Reply:
Ok, it's good we are aligned on the customization.

Now regarding the custom theme, I think I got your logic. I'll try to deploy the custom theme to see what's the difference with light basically. Because only reading the code makes me think there might not be much difference between them but maybe I'm wrong and if it's the case we should probably consider changing `custom` into something better describing the look and feel. I'm thinking about something like `raw`, `minimal` or equivalent. Do you see what I mean?